generators/server/templates/src/main/java/package/repository/_UserRepository.java

```diff
@@ -41,9 +39,11 @@
     Optional<User> findOneByLogin(String login);
     <%_ if (databaseType == 'sql') { _%>
 
-    @Query(value = "select distinct user from User user left join fetch user.authorities",
-        countQuery = "select count(user) from User user")
-    Page<User> findAllWithAuthorities(Pageable pageable);
+    @EntityGraph(attributePaths = "authorities")
+    User findOneWithAuthoritiesById(<%= pkType %> id);
+
+    @EntityGraph(attributePaths = "authorities")
+    Optional<User> findOneWithAuthoritiesByLogin(String login);
```

Comment:
Now, on clean JHipster 3.12.1 cache doesn't seem to work. Each time I get `/api/account`, Hibernate logs ` select user0_.id as id1_4_,...`.  If it doesn't work on the clean application, our solutions will not work too.